### PR TITLE
Titles

### PR DESCRIPTION
**Pull request by Sebastiaanvk**
titles backend for level unlocked titles and achievement unlocked titles

### CODE REVIEW

File: app/models/title.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Title < ApplicationRecord
+  has_many :unlocked_titles, dependent: :destroy
+  has_many :users, through: :unlocked_titles
+  has_one :achievement
```
Comment (JeroenPJ) [inline app/models/title.rb]: Should this not be `optional: :true`?

File: app/models/unlocked_title.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class UnlockedTitle < ApplicationRecord
+  belongs_to :user
+  belongs_to :title
+
+  validates :user_id, uniqueness: { scope: :title_id }
```
Comment (JeroenPJ) [inline app/models/unlocked_title.rb]: Does this mean that there can't be any duplicate user/title couples?

File: db/migrate/20170120191824_add_title_id_to_achievements.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddTitleIdToAchievements < ActiveRecord::Migration[5.0]
+  def change
+    add_reference :achievements, :title, foreign_key: true, default: nil
```
Comment (JeroenPJ) [inline db/migrate/20170120191824_add_title_id_to_achievements.rb]: Is a `optional: :true` required here or not necessary?

File: app/models/title.rb (hunk 1 above)
Comment (Sebastiaanvk) [inline app/models/title.rb]: no not necessary, only the foreign key is default required.

File: app/models/unlocked_title.rb (hunk 2 above)
Comment (Sebastiaanvk) [inline app/models/unlocked_title.rb]: yes

File: db/migrate/20170120191824_add_title_id_to_achievements.rb (hunk 3 above)
Comment (Sebastiaanvk) [inline db/migrate/20170120191824_add_title_id_to_achievements.rb]: no only on the association

**Pull request merged**